### PR TITLE
[ti_mandiant_advantage] Add optional extra fields

### LINKED ISSUE #10650: [ti_mandiant_advantage] Allow including is_publishable, misp, campaigns, and reports
A user requested that the integration retain is_publishable, misp, campaign, and report information in indicator events. Through open-source libraries I can see that the API supports additional query parameters named `include_reports` and `include_campaigns` and `include_misp` to control the latter three.

AFAICT the integration v1.1.1 is already receiving `is_publishable=true/false` and `misp.X=true/false` data. Our pipeline is explicitly dropping those fields. The other two (campaigns and reports) are not in the responses.

For example:

> https: //api.intelligence.mandiant.com/v4/indicator?gte_mscore=50&exclude_osint=True&start_epoch=1662870397&limit=1&end_epoch=1668565800&include_reports=True&include_campaigns=True

So let's add three toggles (boolean variables) that default to off that can control these query params:

- Include related reports?
- Include related campaigns?
- Include misp?

The google/mandiant-ti-client appears to have response examples containing a campaign and report so that we can add tests and mappings.

```json
  "campaigns": [
    {
      "id": "campaign--bff76355-4d90-5f1f-b402-565a8fb2ac61",
      "name": "GLOBAL.21.005",
      "title": "Exploitation of CVE-2021-44228 (aka \"Log4Shell\") in Various Products"
    }
  ]
```

```json
  "reports": [
    {
      "audience": [
        "cyber espionage",
        "fusion"
      ],
      "id": "report--2781217d-3b75-5e22-b3f7-8db3e09d2b70",
      "published_date": "2022-05-11T19:53:16.583Z",
      "report_id": "22-00011950",
      "title": "APT29 Targets European Diplomatic Entities with ROOTSAW Dropper and New BEATDROP Variants Using Dropbox and Slack for C&C",
      "type": "Event Coverage/Implication"
    }
  ]
```

### References

- https://github.com/google/mandiant-ti-client/blob/442772838aa5da889d50b3633181fa2d70a7c0ec/mandiant_threatintel/threat_intel_client.py#L698-L708

- https://github.com/google/mandiant-ti-client/blob/442772838aa5da889d50b3633181fa2d70a7c0ec/tests/fixtures/IndicatorsClient/test_get_indicator_list_multiple_pages_with_parameters.yaml#L82

### PR DESCRIPTION
**Pull request by chrisberkhout**
## Proposed commit message

```
[ti_mandiant_advantage] Add optional extra fields

Add options for requesting the misp, reports and campaigns fields, and
the relevant fields and pipeline logic to handle them.

Before the misp fields were not being requested, but seem to have been
returned anyway. Now the default is to explicitly request them.
```

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Closes #10650

### CODE REVIEW

File: packages/ti_mandiant_advantage/data_stream/threat_intelligence/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -309,14 +309,26 @@ processors:
...
+      def negative = new ArrayList();
+      for (entry in ctx.json.misp.entrySet()) {
+        if (entry.getValue()) {
+          positive.add(entry.getKey());
+        } else {
+          negative.add(entry.getKey());
+        }
+      }
+      ctx.json["misp_warning_list_hits"] = positive;
+      ctx.json["misp_warning_list_misses"] = negative;
```
Comment (efd6) [inline packages/ti_mandiant_advantage/data_stream/threat_intelligence/elasticsearch/ingest_pipeline/default.yml:327]: For testing purposes, do we need to sort these? The `misp` field is a object, so I imagine that is will be iterated over in a non-deterministic way.

File: packages/ti_mandiant_advantage/data_stream/threat_intelligence/manifest.yml [hunk 2]
```diff
@@ -62,6 +62,30 @@ streams:
         multi: false
         required: true
         show_user: false
+      - name: include_misp
+        required: true
+        show_user: true
+        title: Include MISP fields
+        description: These fields show whether the indicator was found in MISP warninglists.
```
Comment (efd6) [inline packages/ti_mandiant_advantage/data_stream/threat_intelligence/manifest.yml:69]:
```suggestion
        description: These fields show whether the indicator was found in MISP warning lists.
```
?

File: packages/ti_mandiant_advantage/data_stream/threat_intelligence/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (chrisberkhout) [reply packages/ti_mandiant_advantage/data_stream/threat_intelligence/elasticsearch/ingest_pipeline/default.yml:327]: Fixed.

Comment (chrisberkhout) [reply packages/ti_mandiant_advantage/data_stream/threat_intelligence/elasticsearch/ingest_pipeline/default.yml:327]: Good catch!

**Review by efd6 — APPROVED** [review]

**Pull request merged**